### PR TITLE
Knot.get unapproved images

### LINKED ISSUE #202: Add a new route that retrieves all unapproved site images
**Why**
- in allowing admins to approve/reject site images, admins can approve or reject them on the admin page somewhere

**What to do**
- add a route to grab all the unapproved images in `ProtectedSiteProcessorImpl.java`
- See [the API reference](https://docs.c4cneu.com/sftt/sftt-api-spec/#get-unapproved-images-admin-only) for more details
- in addition, --> 3 statuses for the images (rejected) --> delete eventually

### PR DESCRIPTION
**Pull request by nalinikantheti**
## Why

Resolves #202

adds a way for devs to get all unapproved images

## This PR

retrieves all images from SITE_IMAGES table with SUBMITTED approval status, so adding another enum in the future may break

## Verification Steps

postman check for http://localhost:8081/api/v1/protected/sites/unapproved_images

### CODE REVIEW

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java
Comment (huang0h) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:885]: ⛏️ Can we replace the raw "SUBMITTED" string with `ImageApprovalStatus.SUBMITTED.getApprovalStatus()` ? It's a bit longer, but this way, if we ever want to change how we represent an image pending approval, we can just change it once in the enum instead of changing a whole bunch of files.

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java
Comment (huang0h) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:896]: ⛏️ make sure to remove any prints before you merge!

File: api/src/main/java/com/codeforcommunity/rest/subrouter/ProtectedSiteRouter.java
Comment (huang0h) [inline api/src/main/java/com/codeforcommunity/rest/subrouter/ProtectedSiteRouter.java:428]: ⛏️ We don't actually need this step of converting `SiteEntryImages` to `JsonObjects` to return JSON at the end - calling `JsonObject.mapFrom()` takes care of converting regular java objects into JSON for us

**Review by huang0h — COMMENTED** [review]
Great work on this ticket as well! Just one more quick comment:

**Comment by CerberusLatrans** [comment]
changed to enum method call and removed json conversion, tested

**Review by huang0h — APPROVED** [review]
lgtm!

File: api/src/main/java/com/codeforcommunity/rest/subrouter/ProtectedSiteRouter.java
Comment (SurabhiKeesara) [inline api/src/main/java/com/codeforcommunity/rest/subrouter/ProtectedSiteRouter.java:429]: Just clarifying question - does this return a list of base64 string images?

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java
Comment (SurabhiKeesara) [inline service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:888]: Is it possible to pull out the SiteEntryImage creation, or the Users SQL query, into a helper method? This makes sense, but is a bit nested to read

File: service/src/main/java/com/codeforcommunity/requester/S3Requester.java
Comment (SurabhiKeesara) [inline service/src/main/java/com/codeforcommunity/requester/S3Requester.java:82]: This is nit picky, just try to remove these extra file changes

**Review by SurabhiKeesara — APPROVED** [review]
looks great!

**Review by CiciLing — APPROVED** [review]
Looked through it with Surabhi. This looks great to me!

File: api/src/main/java/com/codeforcommunity/rest/subrouter/ProtectedSiteRouter.java
Comment (CerberusLatrans) [reply api/src/main/java/com/codeforcommunity/rest/subrouter/ProtectedSiteRouter.java:429]: It gets sent in this format which doesn't have base64 encoding but instead a link to s3 image url: 
{
  "images": [
    {
      "imageId": INT,
      "uploaderUsername": STRING,
      "uploadedAt": TIMESTAMP,
      "imageUrl": STRING
    },
    ...
  ]  
}

File: service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java
Comment (CerberusLatrans) [reply service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java:888]: extracted

**Pull request merged**
